sklearnex/preview/ensemble/_forest.py

```diff
@@ -218,13 +310,88 @@ def estimators_(self, estimators):
         # Needed to allow for proper sklearn operation in fallback mode
         self._cached_estimators_ = estimators
 
```

Comment:
_estimators_ has similarity between regression and classification, with a subtle change in pybind11 interface, this can be generalized and moved to BaseForest